snuba/datasets/schema.py

```diff
@@ -51,6 +51,9 @@ def get_local_table_definition(self):
             self._get_local_engine()
         )
 
+    def get_local_drop_table_statement(self):
```

Comment:
I would remove all the DDL statements from the schema if they are now reachable through DDL. Otherwise it will be confusing for the user of the api which way is the right one to get the DDL statements.

right now we have 
client depends on both DDL and schema and DDL depends on schema. 
If this is an intermediate step, all good, but let's figure out where would the "CREATE TABLE IF NOT EXISTS" belong. Whether in the schema or in the DDL.

Reply:
> I would remove all the DDL statements from the schema if they are now reachable through DDL

I thought about that and felt like it belonged in the schema because I was going to make `MaterializedViewSchema` inherit from `TableSchema`. And the command to create tables vs mat views is different.

> client depends on both DDL and schema and DDL depends on schema.

I do agree that it felt a little strange. However, if we want to make an intermediate schema opaque to the dataset, we do need to have the construction of the DDL separate from the creation of the dataset (i.e. the client).

> Otherwise it will be confusing for the user of the api which way is the right one to get the DDL statements.

That’s a good point. I think the responsibility of `TableSchema` and `DDL` are overlapping too much. Looking at the public API of a `TableSchema`, its entire goal is to provide the table name and the create statements. So, I don’t think we need a `DDL`. I think we need a grouping of `TableSchema`s that we can associate with a dataset.

Maybe we could change dataset to just expose the read/write table names and DDL statements (which can be generated by iterating over the schemas), instead of the schema cause I can’t see why a client needs to know about the schema and not just the name.

What do you think?

Reply:
> So, I don’t think we need a DDL I think we need a grouping of TableSchemas that we can associate with a dataset.

I don't think I agree with this. Removing the intermediate step that makes sense of the schemas associated with a dataset is required otherwise we would be relying on the client to load multiple schemas and figuring out the order in which DDL statements have to be applied.

If you want to merge the DDL responsibilities into the dataset I think we can do that in the short term as long as we do not just expose a list of schemas to the client instead of a list of DDL statements ready to be executed.